src/main/java/net/sf/jabref/logic/util/strings/StringUtil.java

```diff
@@ -646,7 +646,14 @@ public static String expandAuthorInitials(String name) {
                 }
                 for (int j = 1; j < names.length; j++) {
                     if (j == 1) {
-                        sb.append(Author.addDotIfAbbreviation(names[j]));
+                        String[] splittedNames = names[j].split(" ");
+                        if (splittedNames.length > 0) {
+                            sb.append(Author.addDotIfAbbreviation(splittedNames[0]));
+                        }
+                        for (int k = 1; k < splittedNames.length; k++) {
+                            sb.append(' ');
+                            sb.append(Author.addDotIfAbbreviation(splittedNames[k]));
+                        }
```

Comment:
@mairdl The code you provided is the same as

```java
List<String> innerNameList = Arrays.asList(splittedNames);
String innerNames  innerNameList.stream().map(n -> Author.addDotIfAbbreviation(n)).collect(Collectors.joining(" "));
sb.append(innerNames);
```

(untested, please double check if it really works :innocent:)

This is much shorter than your code, probably more readable (if one is used to Java8) and less error prone.

See also http://stackoverflow.com/a/22577565/873282